README.md
# AnimatedSnackBar

A Flutter package to show beautiful animated snackbars directly using overlay.

## Features

- Material-UI

![](https://github.com/JulyWitch/animated_snack_bar/blob/master/files/material.gif?raw=true)

<img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/material-info.png" height="600"> <img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/material-success.png" height="600"> <img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/material-warning.png" height="600"> <img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/material-error.png" height="600">

- Colorized rectangle

<img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/colorized-info.png" height="600"> <img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/colorized-success.png" height="600"> <img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/colorized-warning.png" height="600"> <img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/colorized-error.png" height="600">

- Dark rectangle

<img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/dark-info.png" height="600"> <img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/dark-success.png" height="600"> <img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/dark-warning.png" height="600"> <img src="https://github.com/JulyWitch/animated_snack_bar/raw/master/files/dark-error.png" height="600">

## Getting started

Add package to pubspec.yaml

```bash
flutter pub add animated_snack_bar
```

Import the package

```dart
import 'package:animated_snack_bar/animated_snack_bar.dart';
```

## Usage

- Show material ui snackbar

```dart
AnimatedSnackBar.material(
    'This a snackbar with info type',
    type: AnimatedSnackBarType.info,
).show(context);
```

- Show colorized rectangle snackbar

```dart
AnimatedSnackBar.rectangle(
'Success',
  'This is a success snack bar',
  type: AnimatedSnackBarType.success,
  brightness: Brightness.light,
).show(
  context,
);
```

- Show dark rectangle snackbar

```dart
AnimatedSnackBar.rectangle(
'Success',
  'This is a success snack bar',
  type: AnimatedSnackBarType.success,
  brightness: Brightness.dark,
).show(context);
```

- Remove a snackbar

```dart
final snackbar = AnimatedSnackBar(
    builder: ((context) {
        return Container(
            padding: const EdgeInsets.all(8),
            color: Colors.amber,
            height: 50,
            child: const Text('A custom snackbar'),
        );
    }),
);
snackbar.show(context);

snackbar.remove()
```

- Remove all snackbars

```dart
AnimatedSnackBar.removeAll();
```

- Show

- Show a custom snackbar

```dart
AnimatedSnackBar(
    builder: ((context) {
        return Container(
            padding: const EdgeInsets.all(8),
            color: Colors.amber,
            height: 50,
            child: const Text('A custom snackbar'),
        );
    }),
).show(context);
```

- Show a custom snackbar with MaterialAnimatedSnackBar content

```dart
AnimatedSnackBar(
  builder: ((context) {
    return const MaterialAnimatedSnackBar(
      titleText: 'Custom material snackbar ',
      messageText:
          'This a custom material snackbar with info type',
      type: AnimatedSnackBarType.info,
      foregroundColor: Colors.amber,
      titleTextStyle: TextStyle(
        color: Colors.brown,
      ),
    );
  }),
).show(context);
```

- Changning snackbar pushing position

```dart
AnimatedSnackBar.material(
    'This a snackbar with info type',
    type: AnimatedSnackBarType.info,
    mobileSnackBarPosition: MobileSnackBarPosition.bottom, // Position of snackbar on mobile devices
    desktopSnackBarPosition: DesktopSnackBarPosition.topRight, // Position of snackbar on desktop devices
).show(context);
```

- Change snack bar position for mobile

```dart
AnimatedSnackBar.material(
    'This a snackbar with info type',
    type: AnimatedSnackBarType.info,
    mobilePositionSettings: const MobilePositionSettings(
      topOnAppearance: 100,
      // topOnDissapear: 50,
      // bottomOnAppearance: 100,
      // bottomOnDissapear: 50,
      // left: 20,
      // right: 70,
    ),
).show(context);
```

### Multiple snack bars handling

You can pass `snackBarStrategy` as a paramter to determine what should snack bar do
with snackbars which came before it.

- Should it be shown on them like column? use `ColumnSnackBarStrategy`
- Should it remove them? use `RemoveSnackBarStrategy`
- Should it just stack on them? use `StackSnackBarStrategy`


CHANGELOG.md
## 0.4.0

- Breaking change: `AnimatedSnackBar().show(BuildContext context)` is not a `Future` anymore
- Add: `AnimatedSnackBar.removeAll()` method to remove all snack bars
- Fix: Rendering snackbar behind keyboard
- Fix: `AnimatedSnackBar().remove()` method to remove the snack bar
- Fix: Fadeout time when duration is short (Fadeout time is now 1/4 of the duration clamped between 100 ms to 2000 ms)

## 0.3.1

- Add: MobilePositionSettings to change position of snack bar

## 0.3.0

- Add: MultipleSnackBarStrategy
- Fix: Snack bar position in desktop

## 0.2.2

- Fix: Flutter 3 compability

## 0.2.1

- Fix: Readme images
- Fix: Pubspec description length

## 0.2.0

- Add: Determining snack bar position

- Add: ColorizedRectangle snack bars

- Add: Dark Rectangle snack bars

- Add: New icon assets for future releases

## 0.1.0

- Add: Material-UI snack bars


example/lib/main.dart
import 'package:animated_snack_bar/animated_snack_bar.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Animated Snack Bar',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Animated Snack Bar'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: SingleChildScrollView(
        child: SizedBox(
          width: MediaQuery.of(context).size.width,
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              const Padding(
                padding: EdgeInsets.all(20.0),
                child: TextField(),
              ),
              const SizedBox(height: 50),
              ElevatedButton(
                onPressed: () {
                  AnimatedSnackBar.removeAll();
                },
                child: const Text("Remove all"),
              ),
              const SizedBox(height: 40),
              const Text('Material UI'),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: () {
                  AnimatedSnackBar.material(
                    'This a snackbar with info type',
                    type: AnimatedSnackBarType.info,
                    duration: const Duration(seconds: 1),
                    mobilePositionSettings: const MobilePositionSettings(
                      topOnAppearance: 100,
                      // topOnDissapear: 50,
                      // bottomOnAppearance: 100,
                      // bottomOnDissapear: 50,
                      // left: 20,
                      // right: 70,
                    ),
                    mobileSnackBarPosition: MobileSnackBarPosition.top,
                    desktopSnackBarPosition: DesktopSnackBarPosition.bottomLeft,
                  ).show(context);
                },
                child: const Text("Info"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: () {
                  final snackBar = AnimatedSnackBar.material(
                    'This a snackbar with info type and a very very very long text',
                    type: AnimatedSnackBarType.info,
                    mobileSnackBarPosition: MobileSnackBarPosition.bottom,
                    desktopSnackBarPosition: DesktopSnackBarPosition.bottomLeft,
                    snackBarStrategy: RemoveSnackBarStrategy(),
                  );
                  snackBar.show(context);

                  Future.delayed(const Duration(seconds: 2), () {
                    snackBar.remove();
                  });
                },
                child: const Text("Long Info"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: () {
                  AnimatedSnackBar.material(
                    'This a snackbar with error type',
                    type: AnimatedSnackBarType.error,
                    desktopSnackBarPosition: DesktopSnackBarPosition.topCenter,
                  ).show(context);
                },
                child: const Text("Error"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: () {
                  AnimatedSnackBar.material(
                    'This a snackbar with success type',
                    type: AnimatedSnackBarType.success,
                    mobileSnackBarPosition: MobileSnackBarPosition.bottom,
                    desktopSnackBarPosition: DesktopSnackBarPosition.topRight,
                  ).show(context);
                },
                child: const Text("Success"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: () {
                  AnimatedSnackBar.material(
                    'This a snackbar with warning type',
                    type: AnimatedSnackBarType.warning,
                  ).show(context);
                },
                child: const Text("Warning"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: () {
                  AnimatedSnackBar(
                    builder: ((context) {
                      return const MaterialAnimatedSnackBar(
                        titleText: 'Custom material snackbar ',
                        messageText:
                            'This a custom material snackbar with info type',
                        type: AnimatedSnackBarType.info,
                        foregroundColor: Colors.amber,
                        titleTextStyle: TextStyle(
                          color: Colors.brown,
                        ),
                      );
                    }),
                  ).show(context);
                },
                child: const Text("Custom Material"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: () {
                  AnimatedSnackBar(
                    builder: ((context) {
                      return Container(
                        padding: const EdgeInsets.all(8),
                        color: Colors.amber,
                        height: 50,
                        child: const Text('A custom snackbar'),
                      );
                    }),
                  ).show(context);
                },
                child: const Text("Custom"),
              ),
              const SizedBox(height: 10),
              const Text('Colorized Rectangle'),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: (() {
                  AnimatedSnackBar.rectangle(
                    'Info',
                    'This is an info snack bar',
                    type: AnimatedSnackBarType.info,
                    brightness: Brightness.light,
                  ).show(
                    context,
                  );
                }),
                child: const Text("Info"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: (() {
                  AnimatedSnackBar.rectangle(
                    'Error',
                    'This is an error snack bar',
                    type: AnimatedSnackBarType.error,
                    brightness: Brightness.light,
                  ).show(
                    context,
                  );
                }),
                child: const Text("Error"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: (() {
                  AnimatedSnackBar.rectangle(
                    'Success',
                    'This is a success snack bar',
                    type: AnimatedSnackBarType.success,
                    brightness: Brightness.light,
                  ).show(
                    context,
                  );
                }),
                child: const Text("Success"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: (() {
                  AnimatedSnackBar.rectangle(
                    'Warning',
                    'This is a warning snack bar',
                    type: AnimatedSnackBarType.warning,
                    brightness: Brightness.light,
                  ).show(
                    context,
                  );
                }),
                child: const Text("Warning"),
              ),
              const SizedBox(height: 10),
              const Text('Dark Rectangle'),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: (() {
                  AnimatedSnackBar.rectangle(
                    'Info',
                    'This is an info snack bar',
                    type: AnimatedSnackBarType.info,
                    brightness: Brightness.dark,
                  ).show(
                    context,
                  );
                }),
                child: const Text("Info"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: (() {
                  AnimatedSnackBar.rectangle(
                    'Error',
                    'This is an error snack bar',
                    type: AnimatedSnackBarType.error,
                    brightness: Brightness.dark,
                  ).show(
                    context,
                  );
                }),
                child: const Text("Error"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: (() {
                  AnimatedSnackBar.rectangle(
                    'Success',
                    'This is a success snack bar',
                    type: AnimatedSnackBarType.success,
                    brightness: Brightness.dark,
                  ).show(
                    context,
                  );
                }),
                child: const Text("Success"),
              ),
              const SizedBox(height: 10),
              ElevatedButton(
                onPressed: (() {
                  AnimatedSnackBar.rectangle(
                    'Warning',
                    'This is a warning snack bar',
                    type: AnimatedSnackBarType.warning,
                    brightness: Brightness.dark,
                  ).show(
                    context,
                  );
                }),
                child: const Text("Warning"),
              ),
              const SizedBox(height: 50),
            ],
          ),
        ),
      ),
    );
  }
}
